taskcat/testing/manager.py

```diff
@@ -23,16 +22,17 @@ class TestManager:
     Manages the lifecycle of AWS resources while running a test.
     """
 
-    def __init__(self, config: Config):
+    def __init__(self, config: Config, printer: object = None):
         """
         Creates a TestManager that manages the lifecycle of AWS resources while running a test.
         This instance returned has an additonal property test_definiton that is set after the
         run method is called.
 
         :param config: takes a taskcat Config object.
-        """
+        :param printer: an object capable of reporting test progress
+        """  # noqa: C0301
         self.config: Config = config
-        self.minimal_output: bool = True
+        self.printer = printer
```

Comment:
This is better I think. We can't really keep it as None for long though because the current logic actually relies on the TerminalPrinter to block returning when Creating and Deleting stacks. 

We will need to move the recursive status checks to the stacker status method and we will need to create an abstract base class for the printer to set the type properly here in a future PR. 

What I do like about this is in the future it will also be possible to have a json printer, yaml, XML etc. if we want. Also noticed that the logic for generating the report would also best be in the printer class as well.